Source/DafnyServer/CounterExampleGeneration/DafnyModel.cs

```diff
@@ -456,7 +465,7 @@ public class DafnyModel {
         return new DafnyModelType("?");
       }
       var fullName = GetTrueName(typeElement);
-      if (fullName != null && fullName.Length > 7) {
+      if (fullName != null && fullName.Length > 7 && fullName.Substring(0, 7).Equals("Tclass.")) {
```

Comment:
Otherwise, generics could have a weird name, right?

Reply:
Yes, the boogie name of a concrete Dafny type always starts with "Tclass.", whereas a generic type does not